### PR TITLE
RDK-28700:Thunder iCryptography Port to RDK Security API

### PR DESCRIPTION
**Pull request by yashaswini-rk**
Reason for change:SecApi Backend for Icrypto.Compiled Code.

### CODE REVIEW

**Review by sebaszm — CHANGES_REQUESTED** [review]
Please make sure the code compiles, does not crash or leak memory. Please use coherent indenting and white-spacing.

File: Source/cryptography/implementation/SecApi/Cipher.cpp [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            TRACE_L1(_T("Invalid IV length! [%i]"), ivLength);
+        } else if (maxOutputLength < inputLength) {
+            // Note: Pitfall, AES CBC/ECB will use padding
+            TRACE_L1(_T("Too small output buffer, expected: %i bytes"), inputLength);
+            result = (-inputLength);
+        } else {
+            uint8_t* keyBuf = reinterpret_cast<uint8_t*>(ALLOCA(_keyLength));
+            ASSERT(keyBuf != nullptr);
+
+            uint16_t length = _vault->Export(_keyId, _keyLength, keyBuf, true);
```
Comment (sebaszm) [inline Source/cryptography/implementation/SecApi/Cipher.cpp:171]: This surely can't compile, the prototype is different.

**Review by sebaszm — CHANGES_REQUESTED** [review]
Please make sure the code compiles, does not crash or leak memory. Please use coherent indenting and white-spacing.

**Comment by MFransen69** [comment]
As there was no activity for some time I close the PR. Of course please reopen if it is still desired to respond to the requested changes.

**Pull request closed without merging**